heartbeat/security/security.go

```diff
@@ -36,13 +34,7 @@ func init() {
 	// In the context of a container, where users frequently run as root, we follow BEAT_SETUID_AS to setuid/gid
 	// and add capabilities to make this actually run as a regular user. This also helps Node.js in synthetics, which
 	// does not want to run as root. It's also just generally more secure.
-	sysInfo, err := sysinfo.Host()
-	isContainer := false
-	if err == nil && sysInfo.Info().Containerized != nil {
-		isContainer = *sysInfo.Info().Containerized
-	}
-
-	if localUserName := os.Getenv("BEAT_SETUID_AS"); isContainer && localUserName != "" && syscall.Geteuid() == 0 {
+	if localUserName := os.Getenv("BEAT_SETUID_AS"); localUserName != "" && syscall.Geteuid() == 0 {
```

Comment:
Would custom containers get affected by this change which will not have `BEAT_SETUID_AS` set most likely?

Reply:
I forgot the previous change we did here, which was related to dir privileges when running inside elastic-agent as root.

Reply:
It shouldn't. The only case where this is made more permissive is for native deployments that might have specified `BEAT_SETUID_AS` and are running as root, but those are not supported.

Last time we modified the setuid logic to apply only to the node process.